test/routes/add-article.test.ts

```diff
@@ -20,11 +21,12 @@ describe('add article', (): void => {
   it('should return a successful response', async (): Promise<void> => {
     const articles = new InMemoryArticles();
     const response = await makeRequest(createArticle(), undefined, articles);
+    const [list, count] = await Promise.all([all(articles), articles.count()]);
 
     expect(response.status).toBe(201);
     expect(response.get('Location')).toBe('http://example.com/path-to/article-list');
-    expect(await articles.count()).toBe(1);
-    expect([...articles][0]['http://schema.org/name']).toEqual([{ '@value': 'Article' }]);
+    expect(count).toBe(1);
+    expect(list[0]['http://schema.org/name']).toEqual([{ '@value': 'Article' }]);
```

Comment:
I'd actually keep the individual `await`s here to isolate what each one does (eg if the iterator fails for some reason, can be good to know the previous `expect`s were fine).